### ISSUE TITLE
Decompressing Microsoft DOCX Files in Unreal Engine 5 with Libdeflate

### ISSUE THREAD

**Issue by ffreality** [ebiggers/libdeflate#322]
I want to use libdeflate to decompress Microsoft Office files (for example .docx) to further process them (such as read, write and edit) in Unreal Engine.

**So, I used these codes but it returns at `Actual size calculation unsuccessful.` step. (Line 37)**
https://gist.github.com/ffreality/1caf837b21bf94c6118df790c782d633

Comments says, I need to give an approximate size for `out_nbytes_avail` and it will write actual size to `actual_out_nbytes_ret`.
Some libraries such as PDFium use same workflow. first usage of function gives required allocation sizes (we use NULL for out buffers) and second usage does actual operation.

So, I tried same but I couldn't make it work. Also, I tried to allocate an `uint8*` with upper limit and use it with first decompress function but it didn't work, either.

Is there any suggestion for that ?

**Comment by ffreality** [comment]
Line 37's check returns `LIBDEFLATE_BAD_DATA` result. So, can we say libdeflate doesn't support microsoft related formats ? As I know they use libdeflate.

**Comment by mxmlnkn** [comment]
.docx files are zip archives, which might contain many files. libdeflate is for gzip, zlib, deflate streams, which are used to compress single files, even inside zip archives. But you would have to parse the zip file format to find the raw deflate streams for each file and then use libdeflate to decompress those. Note that files inside zip might also be compressed with bzip2, although rarely.

**Comment by tansy** [comment]
How do you parse zip files? In your example it is unclear.

**Comment by ffreality** [comment]
> How do you parse zip files? In your example it is unclear.

After I looked at Microsoft documentations, I thought I could decompress .docx and .pptx files with deflate decompressor. I just didn't integrated any .zip parser. But after mxmlnkn'e comment I learned that I need to use .zip. I haven't looked at it yet, because I am working with async xlnt.

But we have some .zip plugins. I can use and get informations which I want.
